sympy/integrals/transforms.py

```diff
@@ -1014,9 +1018,30 @@ def expand_dirac_delta(expr):
     """
     return _lin_eq2dict(expr, expr.atoms(DiracDelta))
 
+def laplace_transform_replace_ic(expr, f, F, t, s, ic):
+    """
+    Substitutes ...
+    """
+    res = expr.subs(LaplaceTransform(f(t), t, s), F(s))
+    k=0
+    for c in ic:
+        if k == 0:
+            res = res.subs(f(0), c)
+        else:
+            if k == 1:
+                res = res.subs(Subs(Derivative(f(t), t), t, 0), c)
+            else:
+                res = res.subs(Subs(Derivative(f(t), (t, k)), t, 0), c)
+        k+=1
+    return res
+
 @_noconds
 def _laplace_transform(f, t, s_, simplify=True):
-    """ The backend function for Laplace transforms. """
+    """ The backend function for Laplace transforms.
+
+    This backend assumes that the frontend has already split sums
+    such that `f` is to an addition anymore.
```

Comment:
What function should be called if we don't want to make this assumption?

This should ideally be structured so that there is a single function that can compute the Laplace transform using rules or otherwise return `None`. Or perhaps the function should return say `(A, B)` where `A` is a successfully computed part and `B` is the part that could not be handled and interpretation is that `L(f) = A + L(B)`.

Reply:
> What function should be called if we don't want to make this assumption?
> 
> This should ideally be structured so that there is a single function that can compute the Laplace transform using rules or otherwise return `None`. Or perhaps the function should return say `(A, B)` where `A` is a successfully computed part and `B` is the part that could not be handled and interpretation is that `L(f) = A + L(B)`.

I am assuming that normally you would not call `_laplace_transform` yourself, this is only called by `self.doit()`. The partitioning into sum terms is done there, so the comment means "don't remove that functionality from `doit()`"

Reply:
I think that we would also want to call this from the main `integrate` function so that if someone asks for an integral that is equivalent to a Laplace transform then we can use this routine to calculate it. Since the main `laplace_transform` falls back on `integrate` we need a function that just applies these rules but does not fall back on `integrate`.

Reply:
I have written something that should be useful :)